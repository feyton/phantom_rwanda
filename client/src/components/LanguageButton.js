import i18next from 'i18next';
import React, { useEffect, useRef } from 'react';
import getLanguage from '../utils/getLanguage.js';

const LanguageButton = () => {
  const lanRef = useRef();
  const lan = getLanguage();
  const handleChange = (e) => {
    const { value } = e.target;
    i18next.changeLanguage(value);
  };
  useEffect(() => {
    if (lanRef.current) {
      lanRef.current.value = lan;
    }
  }, []);
  return (
    <div>
      <select
        ref={lanRef}
        onChange={(e) => handleChange(e)}
        className="bg-background border outline-none md:ml-6 lg:ml-6 xl:ml-6 mt-5 flex py-2 px-4 text-sm hover:font-bold"
      >
        <option
          value="en"
          className=" border-none bg-background w-full cursor-pointer"
        >
          English
        </option>
        <option
          value="fr"
          className=" border-none bg-background w-full cursor-pointer"
        >
          French
        </option>
        <option
          value="kn"
          className=" border-none bg-background cursor-pointer"
        >
          Kinyarwanda
        </option>
      </select>
    </div>
  );
};

export default LanguageButton;
