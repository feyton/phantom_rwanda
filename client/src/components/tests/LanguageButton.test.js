import React from 'react';
import renderer from 'react-test-renderer';
import LanguageButton from '../LanguageButton.js';

describe('LanguageButton', () => {
  it('should render LanguageButton', () => {
    const LanguageBtn = renderer.create(<LanguageButton />).toJSON();
    expect(LanguageBtn).toMatchSnapshot();
  });
});
