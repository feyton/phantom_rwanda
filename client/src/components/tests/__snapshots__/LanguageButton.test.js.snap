// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageButton should render LanguageButton 1`] = `
<div>
  <select
    className="bg-background border outline-none md:ml-6 lg:ml-6 xl:ml-6 mt-5 flex py-2 px-4 text-sm hover:font-bold"
    onChange={[Function]}
  >
    <option
      className=" border-none bg-background w-full cursor-pointer"
      value="en"
    >
      English
    </option>
    <option
      className=" border-none bg-background w-full cursor-pointer"
      value="fr"
    >
      French
    </option>
    <option
      className=" border-none bg-background cursor-pointer"
      value="kn"
    >
      Kinyarwanda
    </option>
  </select>
</div>
`;
